core/src/apps/zcash/unified_addresses.py

```diff
@@ -0,0 +1,122 @@
+"""
+Implementation of encoding and decoding of Zcash
+unified addresses according to the ZIP-316.
+see: https://zips.z.cash/zip-0316
+"""
+
+from typing import Dict
+
+from trezor.crypto.bech32 import Encoding, bech32_decode, bech32_encode, convertbits
+from trezor.enums import ZcashReceiverTypecode as Typecode
+from trezor.utils import BufferReader, empty_bytearray
+from trezor.wire import DataError
+
+from apps.common.coininfo import CoinInfo
+from apps.common.readers import read_compact_size
+from apps.common.writers import write_bytes_fixed, write_compact_size
+
+from .f4jumble import f4jumble, f4unjumble
+
+
+def receiver_length(typecode: int):
+    """Byte length of a receiver."""
+    if typecode == Typecode.P2PKH:
+        return 20
+    if typecode == Typecode.P2SH:
+        return 20
+    if typecode == Typecode.SAPLING:
+        return 43
+    if typecode == Typecode.ORCHARD:
+        return 43
+    raise ValueError
+
+
+def prefix(coin: CoinInfo):
+    """Prefix for a unified address."""
+    if coin.coin_name == "Zcash":
+        return "u"
+    if coin.coin_name == "Zcash Testnet":
+        return "utest"
+    raise ValueError
+
+
+def padding(hrp: str) -> bytes:
+    assert len(hrp) <= 16
+    return bytes(hrp, "utf8") + bytes(16 - len(hrp))
```

Comment:
uPy supports only UTF-8

```suggestion
    return bytes(hrp) + bytes(16 - len(hrp))
```

Reply:
https://github.com/trezor/trezor-firmware/pull/2398/commits/e54626dcad3d7fa22fc85c337bb44a760a024d99